execution/block-partitioner/src/test_utils.rs

```diff
@@ -79,3 +110,213 @@ pub fn create_signed_p2p_transaction(
     }
     transactions
 }
+
+pub struct P2PBlockGenerator {
+    accounts: Arc<Vec<Mutex<TestAccount>>>,
+}
+
+impl P2PBlockGenerator {
+    pub fn new(num_accounts: usize) -> Self {
+        let accounts = (0..num_accounts)
+            .into_par_iter()
+            .map(|_i| Mutex::new(generate_test_account()))
+            .collect();
+        Self {
+            accounts: Arc::new(accounts),
+        }
+    }
+
+    pub fn rand_block<R>(&self, rng: &mut R, block_size: usize) -> Vec<AnalyzedTransaction>
+    where
+        R: Rng,
+    {
+        (0..block_size)
+            .map(|_| {
+                let indices = rand::seq::index::sample(rng, self.accounts.len(), 2);
+                let receiver = self.accounts[indices.index(1)].lock().unwrap();
+                let mut sender = self.accounts[indices.index(0)].lock().unwrap();
+                create_signed_p2p_transaction(&mut sender, vec![&receiver]).remove(0)
+            })
+            .collect()
+    }
+}
+
+/// Assert partitioner correctness for `ShardedBlockPartitioner` and `V2Partitioner`:
+/// - Transaction set remains the same after partitioning.
+/// - The relative order of the txns from the same sender
+/// - For a cross-shard dependency, the consumer txn always comes after the provider txn in the sharded block.
+/// - Required edge set matches dependency edge set.
+/// - Before the last round, there is no in-round cross-shard dependency.
+///
+/// Also print a summary of the partitioning result.
+#[cfg(test)]
+pub fn verify_partitioner_output(
```

Comment:
nit: this function it pretty long, can you take a look into breaking it down so that its easier to follow?